pftp/client_handler.go

```diff
@@ -157,21 +149,34 @@ func (c *clientHandler) handleCommands() error {
 						if err := c.conn.Close(); err != nil {
 							c.log.err("Network close error")
 						}
-						return errors.New("idle timeout")
+						proxyError <- err
 					}
-					return err
+					proxyError <- err
 				default:
-					return err
+					proxyError <- err
 				}
 			}
+			line <- readLine
+			<-getResponse
+		}
+	}()
 
-			c.log.debug("read from client: %s", line)
-			commandResponse := c.handleCommand(line)
+	for {
+		select {
+		case e := <-proxyError:
+			return e
```

Comment:
Probably goroutine leaks, so we need to finish the goroutine.

Reply:
right. there was a leaks.
now fixed at https://github.com/pyama86/pftp/pull/26/commits/1ea48e6931939a2e9a8ef47743c620d57a32a519